### PR TITLE
Reg: New Test Case of SAI VOQ Counter https://github.com/sonic-net/sonic-utilities/pull/3322

### PR DESCRIPTION
**Pull request by vikshaw-Nokia**
### Description of PR

Summary:
New Test Cases for  SAI_SWITCH_STAT_PACKET_INTEGRITY_DROP counter in show dropcounter counts command and show SAI_QUEUE_STAT_CREDIT_WD_DELETED_PACKETS counters in show queue counter --voq command.
With Reference of https://github.com/sonic-net/sonic-utilities/pull/3322

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [x] 202405

#### What is the motivation for this PR?
New Test Case for New SAI VOQ Counter

#### How did you do it?
For the SAI_SWITCH_STAT_PACKET_INTEGRITY_DROP counter in the show dropcounter counts command:

We introduced CRC errors using the bcmcmd and verified that the PKT_INTEGRITY_ERR counter incremented by using the show dropcounters counts command.

For the SAI_QUEUE_STAT_CREDIT_WD_DELETED_PACKETS counter in the show queue counter --voq command:

We disabled the TX port while Watchdog (WD) was enabled, then generated traffic. As a result, we observed an increase in the Credit-WD-Del/pkts counter.

#### How did you verify/test it?
Verified on a T2 VOQ Chassis with dnx platform

#### Supported testbed topology if it's a new test case?
T2 VOQ Chassis

### Documentation
Refer https://github.com/sonic-net/sonic-utilities/pull/3322

### CODE REVIEW

File: tests/saitests/py3/sai_base_test.py [hunk 1]
```diff
@@ -204,9 +205,10 @@ def sai_thrift_port_tx_enable(
             assert retValue == 0, "enable wd failed on asic '{}' on '{}' with error '{}'".format(
                 self.src_asic_index, self.src_server_ip, retValue)
 
-    def sai_thrift_port_tx_disable(self, client, asic_type, port_list, target='dst', disable_port_by_block_queue=True):
+    def sai_thrift_port_tx_disable(self, client, asic_type, port_list, target='dst', disable_port_by_block_queue=True,
+                                   wd_diable=True):
```
Comment (vmittal-msft) [inline tests/saitests/py3/sai_base_test.py:209]: wd_diable -> wd_disable

File: tests/saitests/py3/sai_base_test.py [hunk 2]
```diff
@@ -204,9 +205,10 @@ def sai_thrift_port_tx_enable(
...
         count = 0
-        if self.platform_asic and self.platform_asic == "broadcom-dnx":
+        if self.platform_asic and self.platform_asic == "broadcom-dnx" and wd_diable:
```
Comment (vmittal-msft) [inline tests/saitests/py3/sai_base_test.py:211]: same as above

File: tests/saitests/py3/sai_qos_tests.py [hunk 3]
```diff
@@ -6052,3 +6052,56 @@ def findFaultySrcDstPair(dscp, queue):
...
+        log_message("test dst_port_id: {}, src_port_id: {},".format(
+            dst_port_id, src_port_id), to_stderr=True)
+        # in case dst_port_id is part of LAG, find out the actual dst port
+        # for given IP parameters
+        dst_port_id = get_rx_port(
+            self, 0, src_port_id, pkt_dst_mac, dst_port_ip, src_port_ip
+        )
+        log_message("actual dst_port_id: {}".format(dst_port_id), to_stderr=True)
+
+        self.sai_thrift_port_tx_disable(self.dst_client, asic_type, [dst_port_id], wd_diable=False)
```
Comment (kenneth-arista) [inline tests/saitests/py3/sai_qos_tests.py:6098]: Why is the disabling of the hardware watchdog skipped here?

**Comment by kenneth-arista** [comment]
@ysmanman fyi

File: tests/saitests/py3/sai_qos_tests.py (hunk 3 above)
Comment (vikshaw-Nokia) [reply tests/saitests/py3/sai_qos_tests.py:6098]: @kenneth-arista 
The SAI_QUEUE_STAT_CREDIT_WD_DELETED_PACKETS reads the packets deleted by the WD timer. ( The packets which are queued in the Voq, but did not get the credit from the egress and periodic WD timer deletes these packets which are stuck in VOQ). If we disable the WD timer, then the packets in the VOQ will not be deleted and SAI_QUEUE_STAT_CREDIT_WD_DELETED_PACKETS will always return 0.

File: tests/saitests/py3/sai_base_test.py (hunk 1 above)
Comment (vikshaw-Nokia) [reply tests/saitests/py3/sai_base_test.py]: @vmittal-msft Have made the changes

**Comment by vikshaw-Nokia** [comment]
The Pre commit Check is failing because of the below issue which is contradicting to each other.
Please ignore
image

File: tests/voq/files/CRC_ERROR_SUP [hunk 4]
```diff
@@ -0,0 +1,4 @@
+"log off; rm /tmp/esmode; log file=/tmp/esmode quiet=yes"
```
Comment (vmittal-msft) [inline tests/voq/files/CRC_ERROR_SUP:1]: Can we avoid having brcm code here ? is this possible to achieve this via SAI APIs ?

File: tests/voq/files/CRC_Error_LC [hunk 5]
```diff
@@ -0,0 +1,10 @@
+"port enable eth yes"
```
Comment (vmittal-msft) [inline tests/voq/files/CRC_Error_LC:1]: We shouldn't have CINT code or bcm calls here. Please see if this can be achieved via SAI API.

Comment (vikshaw-Nokia) [reply tests/voq/files/CRC_Error_LC:1]: Hello @vmittal-msft ,
At the moment, there is no API available to accomplish this task. We can proceed with the enhancement once Broadcom introduces support for it.

**Comment by vmittal-msft** [comment]
Please update PR subject to indicate the purpose as it makes it easy to follow. Also. please check with BRCM regarding how this can be achieved w/o cint/brcm shell calls.

**Comment by vikshaw-Nokia** [comment]
CSP is opened for requesting the feature

**Comment by rlhui** [comment]
@vikshaw-Nokia  - what exactly is blocking this test currently?  Can we differentiate improvement vs issues?

**Comment by vikshaw-Nokia** [comment]
As discussed, changing the methodology to test fir counters

**Comment by vikshaw-Nokia** [comment]
New PR Created https://github.com/sonic-net/sonic-mgmt/pull/16222

**Pull request closed without merging**